inc/tcpServer.hpp

```diff
@@ -14,23 +14,52 @@
  */
 class TcpServer {
    public:
-	TcpServer(std::string ip, uint32_t port) : acceptor(context, boost::asio::ip::tcp::endpoint(boost::asio::ip::address_v4::from_string(ip), port)), socket(context) {
+	TcpServer(std::string ip, uint32_t port) :
+	acceptor(context, boost::asio::ip::tcp::endpoint(boost::asio::ip::address_v4::from_string(ip), port)),
+	socket(context),
+	acceptIncoming(true) {
```

Comment:
I added a test for this to ensure that this will always fail when trying to call acceptConnection on a native handle, as well as added a specific throw to ensure a non-async throw will be used. Not required but more responsive and easier to test.